crates/compiler/load/tests/test_reporting.rs

```diff
@@ -10916,7 +10916,43 @@ In roc, functions are always written as a lambda, like{}
     6│>          c? 456
     7│       }
     
-    Note: Record builders can only have required values for their fields.
+    Record builders can only have required values for their fields.
+    "#
+    );
+
+    // CalledVia::RecordBuilder => {
+    //     alloc.concat([
+    //         alloc.note(""),
+    //         alloc.reflow("Record builders need a mapper function before the "),
+    //         alloc.keyword("<-"),
+    //         alloc.reflow(" to combine fields together with.")
+    //     ])
+    // }
+    // _ => {
+    //     alloc.reflow("Are there any missing commas? Or missing parentheses?")
+
+    test_report!(
+        record_builder_with_non_function_mapper,
+        indoc!(
+            r#"
+            xyz = "abc"
+
+            { xyz <-
+                b: 123,
+                c: 456
+            }
+            "#
+        ),
+        @r#"
+    ── TOO MANY ARGS in /code/proj/Main.roc ────────────────────────────────────────
+
+    The `xyz` value is not a function, but it was given 3 arguments:
+
+    6│      { xyz <-
+              ^^^
+
+    Note: Record builders need a mapper function before the `<-` to combine
+    fields together with.
```

Comment:
I'd simplify this a bit. Using the word "mapper" might confuse people if they think it's a common term and they try to google it.

```suggestion
    Note: Record builders need a function before the `<-` to combine fields.
```
Once we update the tutorial, I think we should include a link to the record builders section here.